cmd/faucet/faucet.go

```diff
@@ -684,48 +689,99 @@ func sendSuccess(conn *websocket.Conn, msg string) error {
 }
 
 // authTwitter tries to authenticate a faucet request using Twitter posts, returning
-// the username, avatar URL and Ethereum address to fund on success.
-func authTwitter(url string) (string, string, common.Address, error) {
+// the uniqueness identifier (user id/username), username, avatar URL and Ethereum address to fund on success.
+func authTwitter(url string, token string) (string, string, string, common.Address, error) {
 	// Ensure the user specified a meaningful URL, no fancy nonsense
 	parts := strings.Split(url, "/")
 	if len(parts) < 4 || parts[len(parts)-2] != "status" {
 		//lint:ignore ST1005 This error is to be displayed in the browser
-		return "", "", common.Address{}, errors.New("Invalid Twitter status URL")
+		return "", "", "", common.Address{}, errors.New("Invalid Twitter status URL")
+	}
+
+	// Twitter's API isn't really friendly with direct links.
+	// It is restricted to 300 queries / 15 minute with an app api key.
+	// Anything more will require read only authorization from the users and that we want to avoid.
+
+	// If twitter bearer token is provided, use the twitter api
+	if token != "" {
+		// Strip any query parameters from the tweet id
+		tweetID := strings.Split(parts[len(parts)-1], "?")[0]
+
+		// Query the tweet details from Twitter
+		url := "https://api.twitter.com/2/tweets/" + tweetID + "?expansions=author_id&user.fields=profile_image_url"
+		req, err := http.NewRequest("GET", url, nil)
+		if err != nil {
+			return "", "", "", common.Address{}, err
+		}
+		req.Header.Set("Authorization", "Bearer "+token)
```

Comment:
same nitpick here, fmt.Sprintf is the the more proper way to interpolate strings